### DISCUSSION TITLE
Is there any way to cycle through nothing, to bullet point, to all the other marks?

### DISCUSSION THREAD

**Discussion by juanbretti** [ebullient/obsidian-task-collector#423] [Q&A]
Hello,
I would like to know if there is any way to cycle through nothing, to bullet point, to all the other marks?
I mean, for example I have a simple text:

```
list item 1
list item 2
```

Then, I want a hotkey that adds the `-`:

```
- list item 1
- list item 2
```

And then, the same hotkey, to cycle through the other marks, e.g., `[x]`:

```
- [x] list item 1
- [x] list item 2
```

My current solution is to have different hotkeys for:
- Cycle bullet/checkbox
- Task Collector (TC): Mark with next

Thank you!
... and great plugin!

**Comment by ebullient** [comment]
There is a setting that would let you go directly from plain list item to marked list item through the modal. You just can't then cycle back to a regular list item.

I haven't thought about going back to a plain list item... I use the regular Obsidian hotkey for that. Not sure how I would indicate (in config) that you want the cycle to include no-longer-a-task. I don't know how commonly used that configuration would be.

**Reply by juanbretti** [reply]
Thank you for your response.

I am able to convert from plain text to list-item, when using mouse-right-click and select `(TC) Mark Task`

![image](https://github.com/ebullient/obsidian-task-collector/assets/10165911/2a4c0d09-c63d-43ba-9ba2-a7cfdda0f160)

... but **not** by using the hotkey `Task Collector (TC): Mark with next`. I mean, when I press `Shift+Ctrl+Enter`, does not convert from plain to list-item. Also, when I right-click and select `(TC): Mark with next`, it does not convert from plain to list-item.

I have my `hotkeys.json` set like:

```json
  "editor:cycle-list-checklist": [
    {
      "modifiers": [
        "Mod"
      ],
      "key": "Enter"
    }
  ]
  "obsidian-task-collector:task-collector-mark-next": [
    {
      "modifiers": [
        "Mod",
        "Shift"
      ],
      "key": "Enter"
    }
  ]
```

And I have this option enabled, `Convert non-list lines`:

![image](https://github.com/ebullient/obsidian-task-collector/assets/10165911/2e67dd3f-c987-4009-9c64-5ec4c39047ed)

This is my full `data.json`:

```json
{
  "groups": {
    "default": {
      "name": "default",
      "marks": " !\"*/<?ISbcdfiklpuw",
      "complete": false,
      "removeExpr": "",
      "appendDateFormat": "",
      "registerCommand": false,
      "useContextMenu": false,
      "collection": {
        "areaHeading": "## Log",
        "removeCheckbox": false
      }
    },
    "complete": {
      "name": "complete",
      "marks": "->x",
      "complete": true,
      "removeExpr": "",
      "appendDateFormat": "✅ YYYY-MM-DD",
      "registerCommand": false,
      "useContextMenu": false,
      "collection": {
        "areaHeading": "## Log",
        "removeCheckbox": false
      }
    }
  },
  "markCycle": " x->",
  "collectionEnabled": false,
  "previewClickModal": true,
  "contextMenu": {
    "markTask": true,
    "resetTask": false,
    "resetAllTasks": false,
    "collectTasks": false
  },
  "debug": false,
  "convertEmptyLines": true,
  "hideNotifications": false,
  "skipSectionMatch": "",
  "version": {
    "major": 1,
    "minor": 0,
    "patch": 16
  }
}
```

Is this behavior expected?
Thanks again, and great plugin.

**Comment by ebullient** [comment]
So. Mark with Next is _not_ honoring the convert non-list lines setting?

Is it only the "on-ramp" that you're missing? (not-a-task to first-task-in-cycle?)

**Reply by juanbretti** [reply]
Yes. Exactly that. 
Looks like "Mark with Next" is not converting from plain-text to list-item.
Should I change something on my settings? Or is this an expected behavior?

I made a small change on a previous post.

With the action `Cycle bullet/checkbox`, I can make the switch from plain-text to list-item, and back to plain-text.
But this is a core action, so does not cycle the TC marks.

![image](https://github.com/ebullient/obsidian-task-collector/assets/10165911/8232704d-211e-4ef2-93fe-27094b77e9a2)

```json
  "editor:cycle-list-checklist": [
    {
      "modifiers": [
        "Mod"
      ],
      "key": "Enter"
    }
  ]
```

**Comment by ebullient** [comment]
Can we close this as resolved by #456?

**Comment by juanbretti** [comment]
Yes, you can close it. Thanks!
